plugins/chain_plugin/include/eosio/chain_plugin/blockvault_sync_strategy.hpp

```diff
@@ -9,30 +9,49 @@ namespace eosio {
         struct blockvault_sync_strategy : public sync_callback {
             blockvault_sync_strategy(block_vault_interface* blockvault, BP& blockchain_provider, std::function<void()> shutdown,
                                      std::function<bool()> check_shutdown) :
-                _got_snapshot(false), _blockvault(blockvault), _blockchain_provider(blockchain_provider), _shutdown(shutdown), _check_shutdown(check_shutdown) {
+                _startup_run(false), _blockvault(blockvault), _blockchain_provider(blockchain_provider), _shutdown(shutdown), _check_shutdown(check_shutdown) {
+            }
+
+            void run_startup() {
+                if (_blockchain_provider.genesis) {
+                    _blockchain_provider.chain->startup(_shutdown, _check_shutdown, *_blockchain_provider.genesis);
+                }else {
+                    _blockchain_provider.chain->startup(_shutdown, _check_shutdown);
+                }
```

Comment:
Having blockvault_sync_stategy know about genesis seems odd to me. I think this should call a new interface on chain_plugin that knows how to startup correctly.